### ISSUE TITLE
Globe anchor position out of sync after programmatic creation

### ISSUE THREAD

**Issue by r-veenstra** [CesiumGS/cesium-omniverse#600]
Performing some testing in main post https://github.com/CesiumGS/cesium-omniverse/pull/594.  

The globe anchor script featured in https://cesium.com/learn/omniverse/omniverse-placing-objects/ no longer works as expected.  Working fine in https://github.com/CesiumGS/cesium-omniverse/commit/a2dc0ef28ce3be30a99272418ccdeb6217351b20, broken as of https://github.com/CesiumGS/cesium-omniverse/commit/bd9d2d25a77e596f75d64b37fe247a8e58fa01a5

Script copied below for records.

```
import omni.kit.commands
import omni.usd
from pxr import Sdf, UsdGeom, Usd
from cesium.omniverse.api.globe_anchor import anchor_xform_at_path
from asyncio import ensure_future

trees = [
	{"name":"tree_01","latitude":37.79130, "longitude":-122.39032,"height":-28.88791},
	{"name":"tree_02","latitude":37.79114, "longitude":-122.39018,"height":-28.88791},
	{"name":"tree_03","latitude":37.79147, "longitude":-122.39042,"height":-28.88791},
	{"name":"tree_04","latitude":37.79167, "longitude":-122.39053,"height":-28.88791},
]

context = omni.usd.get_context()
stage = context.get_stage()

async def add_trees():
	for t in trees:
	
		# Create the prims name
		payload_prim_path = "/World/"+t["name"]

		# Create the prim for the payload
		payload_prim: Usd.Prim = UsdGeom.Xform.Define(stage, Sdf.Path(payload_prim_path)).GetPrim()
		
		# Add the tree payload to the prim
		omni.kit.commands.execute("AddPayload",
			stage=stage,
			prim_path = payload_prim_path,
			payload=Sdf.Payload(
				assetPath = "http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Vegetation/Plant_Tropical/Cuban_Royal_Palm.usd",
				primPath = Sdf.Path.emptyPath
			)
		)

		# Add a Cesium Globe Anchor
		anchor_xform_at_path(payload_prim_path)
		
		# Wait a frame to allow notifications to complete
		await omni.kit.app.get_app().next_update_async()
		
		# Set the latitude, longitude and height for the tree
		payload_prim.GetAttribute("cesium:anchor:geographicCoordinates").Set(
			(t["latitude"], t["longitude"], t["height"])
		)

ensure_future(add_trees())
```

As of https://github.com/CesiumGS/cesium-omniverse/pull/594, the tree objects are situated at 0,0,0, instead of being located at the correct location based on the globe anchor lat/lon/height.

Each globe anchor appears to have the correct lat/lon/height set in the prim properties window, but this hasn't translated to the correct location on the globe.  If the user manually updates one of the values, it will sync correctly.  Video below.

https://github.com/CesiumGS/cesium-omniverse/assets/123468416/dc054e7e-854c-4d6a-be14-f7a98ba347e4

If I remove `await omni.kit.app.get_app().next_update_async()`, then the trees are at the correct location, however they are scaled and rotated incorrectly, which appears to be the same issue as https://github.com/CesiumGS/cesium-omniverse/issues/539

![Screenshot_5](https://github.com/CesiumGS/cesium-omniverse/assets/123468416/58f0a023-9a35-4426-b876-3b7d0530f9ce)

Likewise, updating `anchor_xform_at_path(payload_prim_path)` to `anchor_xform_at_path(payload_prim_path,t["latitude"], t["longitude"], t["height"])` and removing the subsequent `payload_prim.GetAttribute("cesium:anchor:geographicCoordinates").Set()` results in the same incorrect scale and rotation outcome.

**Comment by lilleyse** [comment]
> If I remove `await omni.kit.app.get_app().next_update_async()`, then the trees are at the correct location, however they are scaled and rotated incorrectly, which appears to be the same issue as https://github.com/CesiumGS/cesium-omniverse/issues/539

That's good at least. `await omni.kit.app.get_app().next_update_async()` was the part I was hoping wouldn't be needed after https://github.com/CesiumGS/cesium-omniverse/pull/594. I'll still need to investigate the other bugs though.

**Comment by lilleyse** [comment]
Confirmed that this is fixed in #626. I had to modify the script a bit since `anchor_xform_at_path` was removed.

```python
import omni.kit.commands
import omni.usd
from pxr import Sdf, UsdGeom, Usd
from asyncio import ensure_future
from cesium.usd.plugins.CesiumUsdSchemas import (
    GlobeAnchorAPI as CesiumGlobeAnchorAPI
)

CESIUM_GEOREFERENCE_PRIM_PATH = "/CesiumGeoreference"

trees = [
    {"name":"tree_01","latitude":37.79130, "longitude":-122.39032,"height":-28.88791},
    {"name":"tree_02","latitude":37.79114, "longitude":-122.39018,"height":-28.88791},
    {"name":"tree_03","latitude":37.79147, "longitude":-122.39042,"height":-28.88791},
    {"name":"tree_04","latitude":37.79167, "longitude":-122.39053,"height":-28.88791},
]

context = omni.usd.get_context()
stage = context.get_stage()

async def add_trees():
    for t in trees:
        # Create the prims name
        payload_prim_path = "/World/"+t["name"]

        # Create the prim for the payload
        payload_prim: Usd.Prim = UsdGeom.Xform.Define(stage, Sdf.Path(payload_prim_path)).GetPrim()
        
        # Add the tree payload to the prim
        omni.kit.commands.execute("AddPayload",
            stage=stage,
            prim_path = payload_prim_path,
            payload=Sdf.Payload(
                assetPath = "http://omniverse-content-production.s3-us-west-2.amazonaws.com/Assets/Vegetation/Plant_Tropical/Cuban_Royal_Palm.usd",
                primPath = Sdf.Path.emptyPath
            )
        )

        # Add a Cesium Globe Anchor
        globe_anchor = CesiumGlobeAnchorAPI.Apply(payload_prim)
        globe_anchor.GetGeoreferenceBindingRel().AddTarget(CESIUM_GEOREFERENCE_PRIM_PATH)

        # Wait a frame to allow notifications to complete
        await omni.kit.app.get_app().next_update_async()

        # Set the latitude, longitude and height for the tree
        globe_anchor.GetGeographicCoordinateAttr().Set((t["latitude"], t["longitude"], t["height"]))

ensure_future(add_trees())
```


Technically `next_update_async()` isn't needed anymore and the correct way to add globe anchors is in https://github.com/CesiumGS/cesium-omniverse/issues/539#issuecomment-1908897209. But still, this shows that you can set globe anchor coordinates after initialization and it updates correctly. You can also assign the georeference in any order and it works.

I'm seeing the same orientation issue as https://github.com/CesiumGS/cesium-omniverse/issues/539#issuecomment-1908897209, which is unsurprising.

![image](https://github.com/CesiumGS/cesium-omniverse/assets/915398/51b5b9fa-f3ab-4e2e-86fa-b2b50891417a)
